tsfresh/feature_selection/selection.py

```diff
@@ -124,7 +128,20 @@ def select_features(X, y, test_for_binary_target_binary_feature=defaults.TEST_FO
 
         y = pd.Series(y, index=X.index)
 
-    df_bh = check_fs_sig_bh(X, y, n_jobs, chunksize, fdr_level, hypotheses_independent,
-                            test_for_binary_target_real_feature)
-
-    return X.loc[:, df_bh[df_bh.rejected].Feature]
+    if multiclass is True:
+        unique_labels = y.unique()
+        if len(unique_labels) < 3:
+            raise RuntimeError("Cannot perform multiclass selection on binary target")
```

Comment:
I don't like the idea of having two codepaths for binary classification. But you are right, this might just happen during cross validation. I'll remove the check.

However, this got me thinking.. Currently, we implicitly assume classification if the target is binary, e.g. `len(set(y)) == 2`. What about ditching that too?
Instead of the `multiclass` or `is_multiclass` parameter, we could have a `selection` parameter that defaults to `'regression'` but could be set to `'classification'` instead. Then, if `selection='regression'` and the target consists of `int` only, we produce a message proposing to set `selection='classification'`.